src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -634,19 +674,9 @@ mongoc_topology_rescan_srv (mongoc_topology_t *topology)
    bool ret;
 
    MONGOC_DEBUG_ASSERT (COMMON_PREFIX (mutex_is_locked) (&topology->mutex));
-
-   if ((topology->description.type != MONGOC_TOPOLOGY_SHARDED) &&
-       (topology->description.type != MONGOC_TOPOLOGY_UNKNOWN)) {
-      /* Only perform rescan for sharded topology. */
-      return;
```

Comment:
This was the cause of the bug, if connecting to a replica set with an SRV string, the polling thread `srv_polling_run` would repeatedly attempt to scan.

Reply:
If we copied the deleted checks (or a call to `mongoc_topology_should_rescan_srv`) to `_mongoc_topology_background_monitoring_start` or `_start_scanner_if_needed` we could leave these checks in here too. That way there's no chance that the user (or one of us) would forget to assert the preconditions are met.